src/App.js

```diff
@@ -0,0 +1,242 @@
+      });
+  };
+
+  const createNewBoard = (newBoard) => {
+    axios
+      .post(`${kBaseUrl}/boards`, newBoard)
+      .then(() => getBoardListDropdown())
+      .then(() => setBoardOption(newBoard.title))
+      .then(() => setChosenBoardData({ cards: [] }))
+      .catch((error) => {
+        console.log(error);
+        throw new Error("Couldn't create a new board.");
+      });
+  };
+
+  const toggleNewBoardForm = () => {
+    setIsBoardFormVisible(!isBoardFormVisible);
+  };
+
+  useEffect(() => {
+    // initially loads boards
+    getBoardListDropdown();
+  }, []);
+
+  useEffect(() => {
+    if (cardOrder && cardSort) {
+      let sortedCardData;
+      if (cardOrder === "Ascending") {
+        if (cardSort === "ID") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.id > b.id ? 1 : -1
+          );
+        } else if (cardSort === "Likes") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.likes > b.likes ? 1 : -1
+          );
+        } else if (cardSort === "Alphabetically") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.message.toLowerCase() > b.message.toLowerCase() ? 1 : -1
+          );
+        }
+      } else if (cardOrder === "Descending") {
+        if (cardSort === "ID") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.id < b.id ? 1 : -1
+          );
+        } else if (cardSort === "Likes") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.likes < b.likes ? 1 : -1
+          );
+        } else if (cardSort === "Alphabetically") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.message.toLowerCase() < b.message.toLowerCase() ? 1 : -1
+          );
+        }
+      }
+      setSortedData([...sortedCardData]);
+    } else {
+      setSortedData(chosenBoardData.cards);
+    }
+  }, [cardOrder, cardSort, boardOption, chosenBoardData]);
+
+  useEffect(() => {
+    if (boards) {
+      for (const board of boards) {
+        if (board.title === boardOption) {
+          setChosenBoardData(board);
+        }
+      }
+    }
+  }, [boardOption, boards]);
+
+  // Get updated board data for selected board and set board
+  const getSelectedBoardData = (boardId) => {
+    axios.get(`${kBaseUrl}/boards/${boardId}`).then((response) => {
+      setChosenBoardData({
+        cards: response.data.cards,
+      });
+    });
+  };
+
+  const addNewCard = (newMessage) => {
+    let boardId;
+    for (const board of boards) {
+      if (board.title === boardOption) {
+        boardId = board.id;
+      }
+    }
+    const requestBody = {
+      message: newMessage,
+      board_id: boardId,
+      likes: 0,
+    };
+    axios
+      .post(`${kBaseUrl}/cards`, requestBody)
+      .then((response) => {
+        return cardApiToJson(response.data);
+      })
+      .then(() => getSelectedBoardData(boardId))
```

Comment:
I'm wondering, were y'all seeing issues having more than 1 statement in a `then` block?